parser/src/lib.rs

```diff
@@ -118,6 +138,21 @@ impl Level {
     }
 }
 
+/// Parses an integer at the beginning of `s`.
+///
+/// Returns the integer and remaining text, if `s` started with an integer. Any errors parsing the
+/// number (which we already know only contains digits) are silently ignored.
+fn parse_integer<T: FromStr>(s: &str) -> Option<(&str, usize)> {
+    let start_digits = s
+        .as_bytes()
+        .iter()
+        .copied()
+        .take_while(|b| b.is_ascii_digit())
+        .count();
```

Comment:
Use https://doc.rust-lang.org/std/primitive.str.html#method.chars?

Reply:
Using a character iterator has the potential to be slightly slower, because each 1-4 bytes will be converted into a 4-byte char. Of course who knows what the optimizer will end up converting your code to, you'd have to benchmark to see. However I don't think there is a downside to using a byte iterator, since we only support looking at ascii characters anyway.

Reply:
> Using a character iterator has the potential to be slightly slower, because each 1-4 bytes will be converted into a 4-byte char. Of course who knows what the optimizer will end up converting your code to, you'd have to benchmark to see. However I don't think there is a downside to using a byte iterator, since we only support looking at ascii digits anyway.

Since the behavior is the same in the end I was in favor of `.chars`, since it simplifies the code.  But I wasn't aware of the potential performance overhead, so it sounds good to me.